src/lonlat.rs

```diff
@@ -39,14 +39,18 @@ impl Latitude {
             _ => return Err(AprsError::InvalidLatitude(b.to_owned())),
         };
 
-        let deg = parse_bytes::<u32>(&b[0..2])
-            .ok_or_else(|| AprsError::InvalidLatitude(b.to_owned()))? as f64;
-        let min = parse_bytes::<u32>(&b[2..4])
-            .ok_or_else(|| AprsError::InvalidLatitude(b.to_owned()))? as f64;
-        let min_frac = parse_bytes::<u32>(&b[5..7])
-            .ok_or_else(|| AprsError::InvalidLatitude(b.to_owned()))? as f64;
-
-        let value = deg + min / 60. + min_frac / 6_000.;
+        // Some APRS lats have trailing spaces
+        // This is used to convey ambiguity and is only valid in latitudes
+        // Once we encounter a space, the remainder must be spaces
+        let (deg, only_spaces) = parse_bytes_trailing_spaces(&[b[0], b[1]], false)
+            .ok_or_else(|| AprsError::InvalidLatitude(b.to_owned()))?;
+        let (min, only_spaces) = parse_bytes_trailing_spaces(&[b[2], b[3]], only_spaces)
+            .ok_or_else(|| AprsError::InvalidLatitude(b.to_owned()))?;
+
+        let (min_frac, _) = parse_bytes_trailing_spaces(&[b[5], b[6]], only_spaces)
+            .ok_or_else(|| AprsError::InvalidLatitude(b.to_owned()))?;
+
+        let value = deg as f64 + min as f64 / 60. + min_frac as f64 / 6_000.;
```

Comment:
shouldn't we keep track of the ambiguity somehow? e.g. having a `latitude_precision` field or something like that?

Reply:
That would definitely be good to support in the future. I mainly wanted to be able to decode more packets, even if we don't extract the most amount of information possible.